### PR TITLE
feat: Add GitHub and Google sign-in buttons

### PR DESCRIPTION
**Pull request by sidarth-23**
## Summary
- Add GitHub sign-in button to allow users to sign in with GitHub.
- Add Google sign-in button to allow users to sign in with Google.
- These buttons will authenticate users and set cookies for access and refresh tokens.
- Added new types for Google Sign-In button and Google Sign-In function.
- Updated Google Sign-In button component to use the new interfaces.
- Updated global window declaration for Google Sign-In functionality.

- [x] Tested (Must)
- [ ] Test Case added
- [x] Build Successful (Must)
- [ ] Sufficient Code comments added (Must)
- [x] Attached Screenshots / Videos 
- [x] All Relevant Documents added

### Screenshot
image

## Depends on
#97

### CODE REVIEW

**Review by yogai53 — DISMISSED** [review]

File: apps/admin/components/google-signin.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  // refs
+  const googleSignInButton = useRef<HTMLDivElement>(null);
+  // states
+  const [gsiScriptLoaded, setGsiScriptLoaded] = useState(false);
+
+  const loadScript = useCallback(() => {
+    if (!googleSignInButton.current || gsiScriptLoaded) return;
+    console.log(process.env.GOOGLE_CLIENT_ID!);
+
+    (window as any)?.google?.accounts.id.initialize({
```
Comment (elizabeththomas92) [inline apps/admin/components/google-signin.tsx]: I see this being used at some places repeatedly ,maybe destructure and use directly? 

`const googleId = window?.google?.accounts?.id;`


```
googleId.initialize({
      client_id: clientId,
      callback: handleGoogleSignIn,
    });
```

**Review by elizabeththomas92 — CHANGES_REQUESTED** [review]

File: apps/admin/components/github-signin.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        });
+
+        cookieSetter({
+          access: response.data.tokens.access,
+          refresh: response.data.tokens.refresh,
+        });
+      } else throw Error("Cant find credentials");
+    } catch (err: any) {
+      // TODO: Handle error
+      console.error(err);
```
Comment (yaju09) [inline apps/admin/components/github-signin.tsx]:
```suggestion
      console.error(err);
```
please remove console

File: apps/admin/components/github-signin.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    }
+  };
+  const searchParams = useSearchParams();
+  // states
+  const [loginCallBackURL, setLoginCallBackURL] = useState(undefined);
```
Comment (yaju09) [inline apps/admin/components/github-signin.tsx]:
```suggestion
  const [loginCallBackURL, setLoginCallBackURL] = useState();
```
is undefined is to be given explicitly?

File: apps/admin/components/github-signin.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      handleGitHubSignIn(code);
+    }
+  }, [code, gitCode, handleGitHubSignIn]);
+
+  useEffect(() => {
+    const origin =
+      typeof window !== "undefined" && window.location.origin
+        ? window.location.origin
+        : "";
+    setLoginCallBackURL(`${origin}/` as any);
```
Comment (yaju09) [inline apps/admin/components/github-signin.tsx]:
```suggestion
    setLoginCallBackURL(`${origin}/` as any);
``
is it possible to handle any?`

File: apps/admin/components/google-signin.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  shape?: "rectangular" | "pill" | "circle" | "square";
+  logo_alignment?: "left" | "center";
+  width?: number;
+  local?: string;
+}
+
+export const GoogleSignInButton: FC<Props> = () => {
+  const router = useRouter();
+  const clientId = process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID;
+  const handleGoogleSignIn = async ({ clientId, credential }: any) => {
```
Comment (yaju09) [inline apps/admin/components/google-signin.tsx]:
```suggestion
  const handleGoogleSignIn = async ({ clientId, credential }: any) => {
```
is it possible to address any?

File: apps/admin/components/google-signin.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        }
+      );
+      cookieSetter({
+        access: response.data.tokens.access,
+        refresh: response.data.tokens.refresh,
+      });
+      router.push("/admin/dashboard");
+    } catch (err) {
+      const error = err as AxiosError<TGoogleSignInErrorResponse>;
+      console.error(error);
```
Comment (yaju09) [inline apps/admin/components/google-signin.tsx]:
```suggestion
      console.error(error);
```
is the above console required?

File: apps/admin/components/google-signin.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        {
+          type: "standard",
+          theme: "outline",
+          size: "large",
+          logo_alignment: "center",
+          text: "signin_with",
+        } as GsiButtonConfiguration // customization attributes
+      );
+    } catch (err) {
+      console.log(err);
```
Comment (yaju09) [inline apps/admin/components/google-signin.tsx]:
```suggestion
      console.log(err);
```
is the console required?

**Review by yaju09 — CHANGES_REQUESTED** [review]

File: apps/admin/components/google-signin.tsx (hunk 5 above)
Comment (sidarth-23) [reply apps/admin/components/google-signin.tsx]: I think that is bit difficult for now

**Review by elizabeththomas92 — DISMISSED** [review]

**Review by yaju09 — APPROVED** [review]

**Review by elizabeththomas92 — APPROVED** [review]

**Review by yogai53 — APPROVED** [review]

**Pull request merged**